### ISSUE TITLE
keras.model.predict also needs two inputs (one for sample, and one for label) when used with tf.data together

### ISSUE THREAD

**Issue by monkeyCv** [tensorflow/tensorflow#22757]
I use tf.data.TFRecordDataset as the input of the tf.keras.model. The training and evaluation are OK. When I used the model for inference, I call the function tf.keras.model.prediction. However, it shows me an error 

> Please provide model inputs as a list or tuple of 2 elements: input and target pair.

I check the source code, and found that the function "_standardize_user_data" (line 988) in "./tensorflow/python/keras/engine/training.py" check the input number as the training and evaluation. I think it may be a bug, because when I does prediction, I don't know the true label.

tensorflow version: 1.11.0 gpu.

Thanks for your notice. I am looking forward your reply.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by wt-huang** [comment]
@monkeyCv To make inference, use `tf.keras.model.predict_generation()` instead. `_standardize_user_data `does require you to feed the true label but it is only used during training.

**Comment by monkeyCv** [comment]
@wt-huang Thanks very much. I will try to use `tf.keras.model.predict_generation()`. I am not sure whether the API can recieve the argument `tf.data`, and the document doesnot mention it.
Here I think `tf.keras.model.predict()` calls `_standardize_user_data`, becuase I feed the label, it works well. If I donnot feed the label, it shows an erreor like the figure shows.
![image](https://user-images.githubusercontent.com/38521537/46566548-53c9c200-c953-11e8-8665-0356ce8dffba.png)

my test code:   
```
def test(test_tfrecord,norm_mean,norm_std):
    test_dataset=tf.data.TFRecordDataset(test_tfrecord,num_parallel_reads=4)
    def test_parse_function(example_proto):
        features={'image_raw':tf.FixedLenFeature((),tf.string,default_value='')}
        parsed_features=tf.parse_single_example(example_proto,features)
        image=tf.decode_raw(parsed_features['image_raw'],tf.uint8)
        image=tf.reshape(image,shape=(height,width,channels))
        image=tf.cast(image,tf.float32)
        image=image/255.0
        image=image-norm_mean
        image=image/norm_std
        return image
    test_dataset=test_dataset.map(test_parse_function)
    test_dataset=test_dataset.batch(1)
    
    input_layer=keras.Input(shape=(height,width,channels))
    base_net=myResnet50.ResNet50(input_tensor=input_layer,
                                                  include_top=False,weights='imagenet')
    x=base_net.output
    x=keras.layers.Flatten()(x)
    predictions=keras.layers.Dense(classes,activation='softmax')(x)
    net=keras.models.Model(inputs=input_layer,outputs=predictions)
    net.compile(optimizer=keras.optimizers.Adam(),loss='categorical_crossentropy',metrics=['accuracy'])
    ckpt = tf.train.get_checkpoint_state('../model')
    if not ckpt or   not ckpt.model_checkpoint_path:
        print("No model! Start to train the model!")
    else:
        net.load_weights('../model/mine')
        print("Model is loaded!")
    rst=net.predict(test_dataset,steps=440)
    return rst

```
my platform: win10, anaconda 1.8.7, spyder 3.2.3, tensorflow 1.11.0 gpu, cuda 9 and cudnn 7.3.

**Comment by luoweiyi** [comment]
I have met the same problem when I use model.predict(tf.data), do you solve this problem?
I think it must be a bug after I check the source code,training and evaluate are both ok because we have two input(image and label),but when we do prediction,we only have one input and it goes wrong

**Comment by monkeyCv** [comment]
> I have met the same problem when I use model.predict(tf.data), do you solve this problem?
> I think it must be a bug after I check the source code,training and evaluate are both ok because we have two input(image and label),but when we do prediction,we only have one input and it goes wrong

No. I just input a label like placeholder without any meaning to call the predict program.

**Comment by wt-huang** [comment]
The labels can be omitted when using the model.predict() method. You should be able to use tf.data with model.predict() without getting errors. Alternatively, you can work with numpy array. Make sure to initialize model properly and invoke model.fit() to feed inputs and label.
